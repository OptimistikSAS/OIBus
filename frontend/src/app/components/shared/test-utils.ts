import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
