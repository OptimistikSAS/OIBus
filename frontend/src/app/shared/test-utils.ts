import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
