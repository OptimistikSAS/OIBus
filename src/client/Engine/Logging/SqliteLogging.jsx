import React from 'react'
import PropTypes from 'prop-types'
import OIbForm from '../../components/OIbForm/OIbForm.jsx'
import { minValue, notEmpty } from '../../../services/validation.service'

const schema = { name: 'SQLiteLogging' }
schema.form = {
  level: {
    type: 'OIbSelect',
    md: 3,
    options: ['silly', 'debug', 'info', 'warning', 'error', 'none'],
    defaultValue: 'info',
    help: <div>The level for the sqlite log</div>,
  },
  fileName: {
    type: 'OIbText',
    label: 'File name of sqlite db',
    md: 4,
    valid: notEmpty(),
    defaultValue: '',
    help: <div>The file name of the database file</div>,
  },
  maxSize: {
    type: 'OIbInteger',
    newRow: false,
    md: 2,
    label: 'Database max size',
    valid: minValue(10000),
    defaultValue: 1000000,
    help: <div>Max size of the sqlite database (Byte)</div>,
  },
}

const SqliteLogging = ({ logParameters, onChange }) => (
  <div>
    <h6>SQLite</h6>
    <OIbForm onChange={onChange} schema={schema} name="engine.logParameters.sqliteLog" values={logParameters} />
  </div>
)

SqliteLogging.propTypes = {
  onChange: PropTypes.func.isRequired,
  logParameters: PropTypes.object.isRequired,
}
export default SqliteLogging
