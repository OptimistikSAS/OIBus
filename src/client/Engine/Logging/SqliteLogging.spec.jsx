import React from 'react'
import ReactDOM from 'react-dom'
import { act, Simulate } from 'react-dom/test-utils'

import testConfig from '../../../../tests/testConfig'
import SqliteLogging from './SqliteLogging.jsx'

const onChange = jest.fn()

const mockMath = Object.create(global.Math)
mockMath.random = () => 1
global.Math = mockMath

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

describe('SqliteLogging', () => {
  test('check SqliteLogging', () => {
    act(() => {
      ReactDOM.render(<SqliteLogging
        logParameters={testConfig.engine.logParameters.sqliteLog}
        onChange={() => (1)}
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
  test('check change sqliteLevel to "warning"', () => {
    act(() => {
      ReactDOM.render(<SqliteLogging
        logParameters={testConfig.engine.logParameters.sqliteLog}
        onChange={onChange}
      />, container)
    })
    Simulate.change(document.getElementById('engine.logParameters.sqliteLog.level'), { target: { value: 'warning', selectedIndex: 4 } })
    expect(onChange).toBeCalledWith('engine.logParameters.sqliteLog.level', 'warning', null, null)
    expect(container).toMatchSnapshot()
  })
  test('check change sqliteLog fileName to "new_sqliteFilename"', () => {
    act(() => {
      ReactDOM.render(<SqliteLogging
        logParameters={testConfig.engine.logParameters.sqliteLog}
        onChange={onChange}
      />, container)
    })
    Simulate.change(document.getElementById('engine.logParameters.sqliteLog.fileName'), { target: { value: 'new_sqliteFilename' } })
    expect(onChange).toBeCalledWith('engine.logParameters.sqliteLog.fileName', 'new_sqliteFilename', null)
    expect(container).toMatchSnapshot()
  })
  test('check change sqliteLog maxSize to 10000000', () => {
    act(() => {
      ReactDOM.render(<SqliteLogging
        logParameters={testConfig.engine.logParameters.sqliteLog}
        onChange={onChange}
      />, container)
    })
    Simulate.change(document.getElementById('engine.logParameters.sqliteLog.maxSize'), { target: { value: 10000000 } })
    expect(onChange).toBeCalledWith('engine.logParameters.sqliteLog.maxSize', 10000000, null)
    expect(container).toMatchSnapshot()
  })
})
