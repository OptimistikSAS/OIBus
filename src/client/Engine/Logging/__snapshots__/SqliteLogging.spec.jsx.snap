// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SqliteLogging check SqliteLogging 1`] = `
<div>
  <div>
    <h6>
      SQLite
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.level"
            name="engine.logParameters.sqliteLog.level"
          >
            <option
              value="silly"
            >
              silly
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
            <option
              value="none"
            >
              none
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the sqlite log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.fileName"
          >
            File name of sqlite db
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.fileName"
            name="engine.logParameters.sqliteLog.fileName"
            type="text"
            value="./logs/journal.db"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The file name of the database file
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.maxSize"
          >
            Database max size
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.maxSize"
            name="engine.logParameters.sqliteLog.maxSize"
            type="integer"
            value="50000000"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              Max size of the sqlite database (Byte)
            </div>
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SqliteLogging check change sqliteLevel to "warning" 1`] = `
<div>
  <div>
    <h6>
      SQLite
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.level"
            name="engine.logParameters.sqliteLog.level"
          >
            <option
              value="silly"
            >
              silly
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
            <option
              value="none"
            >
              none
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the sqlite log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.fileName"
          >
            File name of sqlite db
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.fileName"
            name="engine.logParameters.sqliteLog.fileName"
            type="text"
            value="./logs/journal.db"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The file name of the database file
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.maxSize"
          >
            Database max size
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.maxSize"
            name="engine.logParameters.sqliteLog.maxSize"
            type="integer"
            value="50000000"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              Max size of the sqlite database (Byte)
            </div>
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SqliteLogging check change sqliteLog fileName to "new_sqliteFilename" 1`] = `
<div>
  <div>
    <h6>
      SQLite
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.level"
            name="engine.logParameters.sqliteLog.level"
          >
            <option
              value="silly"
            >
              silly
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
            <option
              value="none"
            >
              none
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the sqlite log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.fileName"
          >
            File name of sqlite db
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.fileName"
            name="engine.logParameters.sqliteLog.fileName"
            type="text"
            value="./logs/journal.db"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The file name of the database file
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.maxSize"
          >
            Database max size
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.maxSize"
            name="engine.logParameters.sqliteLog.maxSize"
            type="integer"
            value="50000000"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              Max size of the sqlite database (Byte)
            </div>
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SqliteLogging check change sqliteLog maxSize to 10000000 1`] = `
<div>
  <div>
    <h6>
      SQLite
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.level"
            name="engine.logParameters.sqliteLog.level"
          >
            <option
              value="silly"
            >
              silly
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
            <option
              value="none"
            >
              none
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the sqlite log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.fileName"
          >
            File name of sqlite db
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.fileName"
            name="engine.logParameters.sqliteLog.fileName"
            type="text"
            value="./logs/journal.db"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The file name of the database file
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="engine.logParameters.sqliteLog.maxSize"
          >
            Database max size
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.sqliteLog.maxSize"
            name="engine.logParameters.sqliteLog.maxSize"
            type="integer"
            value="50000000"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              Max size of the sqlite database (Byte)
            </div>
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;
